packages/vue-components/components/profile/update-email/update-email.vue

```diff
@@ -0,0 +1,98 @@
+<template>
+  <generic-form
+    :fields
+    :buttons="{
+      submit: true,
+      reset: false,
+      cancel: false,
+    }"
+    :labels="{
+      submit: t('locokit.components.updateEmailForm.submit'),
+    }"
+    :loading="loading"
+    :message="message"
+    @submit="onSubmit"
+  >
+    <template #top>
+      <div class="mb-4">
+        <p>{{ $t('locokit.components.updateEmailForm.currentEmail') }}</p>
+        <p class="font-bold">{{ user?.email }}</p>
+      </div>
+    </template>
+  </generic-form>
+</template>
+
+<script setup lang="ts">
+import { computed } from 'vue'
+import { useI18n } from 'vue-i18n'
+import { FIELD_COMPONENT, FIELD_TYPE, LocoKitFormField, LocoKitMessage } from '@locokit/definitions'
+import GenericForm from '@/components/commons/generic-form/generic-form.vue'
+
+const { t } = useI18n()
+
+const emit = defineEmits<{
+  /**
+   * Emitted when the submit button has been clicked
+   * and the form has been successfully validated.
+   */
+  (
+    e: 'submit',
+    form: {
+      id: string
+      newEmail: string
+      password: string
+    },
+  ): void
```

Comment:
I think it could be nicer to use the latest way of defining events emitted:
https://vuejs.org/api/sfc-script-setup.html#type-only-props-emit-declarations

```suggestion
  submit: [
    form: {
      id: string
      newEmail: string
      password: string
    }
  ]
```

If you agree, please update other components in this PR.